packages/drizzle-duckdb-wasm/src/dsn.ts

```diff
@@ -0,0 +1,115 @@
+import type { DBStorage } from '@proj-airi/duckdb-wasm'
+
+import { DuckDBAccessMode } from '@duckdb/duckdb-wasm'
+import { DBStorageType } from '@proj-airi/duckdb-wasm'
+
+export interface StructuredDSN {
+  protocol: 'duckdb-wasm:'
+  bundles?: 'import-url'
+  logger?: boolean
+  storage?: DBStorage
+}
+
+export function isLiterallyTrue(value?: string): boolean {
+  return typeof value === 'string' && /^true$/i.test(value)
+}
+
+/**
+ * Parse a DuckDB WASM DSN string into a structured object
+ *
+ * Examples:
+ * - `duckdb-wasm:///` -> In-memory
+ * - `duckdb-wasm:///?bundles=import-url` -> In-memory, use import-URL bundles
+ * - `duckdb-wasm:///?logger=true` -> In-memory, enable logger
+ * - `duckdb-wasm://database.db?storage=origin-private-fs&write=true` -> Origin Private FS, RW, database.db
+ * - `duckdb-wasm:///database.db?storage=origin-private-fs&write=true` -> Origin Private FS, RW, database.db (leading slash is optional)
+ *
+ * @param dsn The DSN string to parse
+ */
+export function parseDSN(dsn: string): StructuredDSN {
+  const structured: StructuredDSN = {
+    protocol: 'duckdb-wasm:',
+  }
+
+  const parsed = new URL(dsn)
+
+  if (!parsed.protocol.startsWith('duckdb-wasm:')) {
```

Comment:
```suggestion
  if (!parsed.scheme.startsWith('duckdb-wasm:')) {
```

Ref: https://developer.mozilla.org/en-US/docs/Web/URI/Reference/Schemes